### PR TITLE
navigate checkbox with arrows/enter and change search focus on '/' hotkey

### LINKED ISSUE #19: Add keyboard shortcuts
- `/` for highlight of the main searchbar
- Return key to select checkboxes or author search (maybe just map to space?)
- arrow to navigate the sidebar (though tabs work now)

The first seems like an easy addition, maybe worth discussing the other two.

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**
closes #19

### CODE REVIEW

File: src/hooks/useCheckboxNavigate.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      const charCode =
+        e.keyCode || (typeof e.which === "number" && e.which) || null;
+
+      switch (charCode) {
+        case 40:
+          // downArrow
+          if (currentCheckboxNavIndex === null) {
+            currentCheckboxNavIndex = 0;
+          } else {
+            // if (multiCheckboxList.length - 1 === currentCheckboxNavIndex) {
```
Comment (adamjonas) [inline src/hooks/useCheckboxNavigate.js:38]: unused code here?

**Comment by Emmanuel-Develops** [comment]
Due to the new eslint rules, I refactored existing components and renamed others.

**Comment by adamjonas** [comment]
I would have preferred to break the eslint into another PR. This is now much harder to review.

**Comment by adamjonas** [comment]
The slash works well but the arrows and enter key have unexpected behaviors.

If I use the arrow key in the text box it toggle between the author and previous prompt.

If I hit return to select a checkbox, return doesn't uncheck it. I need to use space.

https://user-images.githubusercontent.com/755825/213207477-77376762-93be-40de-8b77-6a748703ec63.mov

**Comment by Emmanuel-Develops** [comment]
> The slash works well but the arrows and enter key have unexpected behaviors.
> 
> If I use the arrow key in the text box it toggle between the author and previous prompt.
> 
> If I hit return to select a checkbox, return doesn't uncheck it. I need to use space.
> 
>  Screen.Recording.2023-01-18.at.10.07.03.AM.mov

The enter key unchecks the prompt when the toggle is at the previous prompt.

The current styling specificity has "checked" as higher precedence over "navigation" I'll change that.

What do you think about "CMD/CTRL + K" to search? It's a common hotkey for search. Should we add the key binding along with "/"

**Comment by Emmanuel-Develops** [comment]
https://user-images.githubusercontent.com/62019510/213254898-dd838fa8-8f14-4580-b92e-e3251bf14661.mov

How about this?
I restricted the navigation to within options. 
The filter is focused and active while navigating.
Added previous state memory to remember last checked/unchecked option and continue navigating continues from there

**Comment by adamjonas** [comment]
What's the UI to tell people about the shortcuts you are adding?

**Comment by adamjonas** [comment]
hitting the `/` not only highlights the input but it writes the slash in the input

**Comment by adamjonas** [comment]
I can navigate down to the checkboxes but not back up to the input field

https://user-images.githubusercontent.com/755825/213491363-1f510808-4222-4b68-aea8-74d72d273b53.mov

**Comment by adamjonas** [comment]
Weird behavior when unchecking authors and returning to input box


https://user-images.githubusercontent.com/755825/213492105-312a6086-5925-43f2-8da6-927bec4609c5.mov

**Comment by Emmanuel-Develops** [comment]
> Details

the input field is actively focused and navigation is trapped among the options.
You can search at any point during navigation, it will continue from where you stopped typing and filter the options.
The up and down has a side-effect that changes the position of the input cursor (from beginning to end of the word). I've fixed that.

**Comment by Emmanuel-Develops** [comment]
> Weird behavior when unchecking authors and returning to input box
> 
>  Screen.Recording.2023-01-19.at.11.02.26.AM.mov

I can't recreate this on either chrome or safari. What browser was used to create this?

**Comment by Emmanuel-Develops** [comment]
> What's the UI to tell people about the shortcuts you are adding?

seachbar-keyboard-hint

it could be in the red box as <div><span>💡: </span><span>Search with</span> <kbd>⌘</kbd><kbd>k</kbd> or <kbd>/</kbd></div>

**Comment by adamjonas** [comment]
tACK

If you lose the focus on the facet input box it can get a little funky, but I'm ok with this as a v1.

I'll create an issue on the hotkeys UI

**Comment by adamjonas** [comment]
In the future, let's not wrap unrelated features together in 1 PR. The eslint formatting additions should have been split out on their own.

**Comment by Emmanuel-Develops** [comment]
> In the future, let's not wrap unrelated features together in 1 PR. The eslint formatting additions should have been split out on their own.

Yes I agree. It was meant to be a smaller fix kinda spiraled out. There would be more segregation going forward

**Pull request merged**